fvp.sh

```diff
@@ -36,10 +36,47 @@ if ! docker image inspect "fvp:${FVP_VERSION}" >/dev/null 2>&1; then
     "${DIRNAME}/build.sh"
 fi
 
+# Define the default mounts
+MOUNTS=()
+
+# Mount home directory by default unless explicitly disabled or custom workdir is set
+if [[ "$FVP_DISABLE_HOME_MOUNT" != "true" ]] && [[ -z "$FVP_MAC_WORKDIR" ]]; then
+    echo "Warning: FVPs-on-Mac container is mounting entire home directory. For improved performance and security, consider setting FVP_DISABLE_HOME_MOUNT=true or FVP_MAC_WORKDIR to limit mounted directories." >&2
+    MOUNTS+=("--mount" "type=bind,src=${HOME},dst=${HOME}")
+else
+    MOUNTS+=("--mount" "type=bind,src=${HOME}/.armlm/,dst=${HOME}/.armlm/")
+fi
+
+# Set working directory
+WORKDIR="$(pwd)"
+
+# Handle custom workdir if specified
+if [[ -n "$FVP_MAC_WORKDIR" ]]; then
+    if [[ ! -d "$FVP_MAC_WORKDIR" ]]; then
+        echo "Error: FVP_MAC_WORKDIR path '$FVP_MAC_WORKDIR' does not exist or is not a directory" >&2
+        exit 1
+    fi
+    MOUNTS+=("--mount" "type=bind,src=${FVP_MAC_WORKDIR},dst=${FVP_MAC_WORKDIR}")
+    WORKDIR="$FVP_MAC_WORKDIR"
```

Comment:
Setting `WORKDIR` to the top level of the mount is not equivalent with having `pwd` in a subdirectory of `HOME`.

E.g., I want to restrict the mount to `/home/user/my-project` and then launch an FVP in `/home/user/my-project/some/folder`. Lets say `FVP_MPS2_Cortex-M3 -a app.axf` where the image is `/home/user/my-project/some/folder/app.axf`. This results in an error message because inside of the container the workdir is set to `/home/user/my-project` and FVP can't find the `app.axf`.